scripts/generate_dump

```diff
@@ -1211,39 +1211,42 @@ collect_broadcom() {
     fi
 
     if [ "$bcm_family" == "broadcom-dnx" ]; then
-       save_bcmcmd_all_ns "\"l2 show\"" "l2.summary"
-       save_bcmcmd_all_ns "\"field group list\"" "fpgroup.list.summary"
-       total_fp_groups=34
-       for (( fp_grp=0; fp_grp<$total_fp_groups; fp_grp++ ))
-       do
-         save_bcmcmd_all_ns "\"field group info group=$fp_grp\"" "fpgroup$fp_grp.info.summary"
-       done
-       save_bcmcmd_all_ns "\"dbal table dump table=IPV4_UNICAST_PRIVATE_LPM_FORWARD\"" "l3.ipv4.lpm.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=IPV6_UNICAST_PRIVATE_LPM_FORWARD\"" "l3.ipv6.lpm.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=IPV4_UNICAST_PRIVATE_HOST\"" "l3.ipv4.host.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=IPV6_UNICAST_PRIVATE_HOST\"" "l3.ipv6.host.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=SUPER_FEC_1ST_HIERARCHY\"" "l3.egress.fec.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=ECMP_TABLE\"" "ecmp.table.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=ECMP_GROUP_PROFILE_TABLE\"" "ecmp.group.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=ING_VSI_INFO_DB\"" "ing.vsi.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=L3_MY_MAC_DA_PREFIXES\"" "l3.mymac.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=INGRESS_VLAN_MEMBERSHIP\"" "ing.vlan.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=LOCAL_SBC_IN_LIF_MATCH_INFO_SW\"" "sbc.inlif.summary"
-       save_bcmcmd_all_ns "\"dbal table dump table=SNIF_COMMAND_TABLE\"" "snif.command.summary"
-       save_bcmcmd_all_ns "\"port mgmt dump full\"" "port.mgmt.summary"
-       save_bcmcmd_all_ns "\"tm lag\"" "tm.lag.summary"
-       save_bcmcmd_all_ns "\"pp info fec\"" "pp.fec.summary"
-       save_bcmcmd_all_ns "\"nif sts\"" "nif.sts.summary"
+       switch_type=$(sonic-cfggen -d -v "DEVICE_METADATA['localhost']['switch_type']")
```

Comment:
with this check `if [ $switch_type != "fabric" ]`if the supervisor has no configuration then 'show techsupport` will fail?

Instead of check for `switch_type`, Can we use the flag `is_supervisor` in the `platfrom_env.conf` in hwsku folder to check if the card is supervisor or linecard